c++/src/linked-list.cc

```diff
@@ -174,31 +174,23 @@ class LinkedList {
     }
 
     Node<T>* lookingAt = this->head;
-    Node<T>* nodeToDelete = nullptr;
     Node<T>* previousNode = nullptr;
 
-    while (true) {
+    while (lookingAt->getValue() != value) {
```

Comment:
Starting from scratch, this condition IMO should've been `while (lookingAt != nullptr)`, then from there make the necessary changes within the body, i.e. remove `if (lookingAt == nullptr)` and promote the `else if` to `if`.